actions/entrypoint.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/sh -l
+
+sourceDirectory=$1
+platform=$2
+platformVersion=$3
+
+echo
+
+if [ -n "${sourceDirectory}" ]
+then
+    sourceDirectory="$PWD/$sourceDirectory"
+    echo "Relative path to source directory provided -- the following directory will be built: '${sourceDirectory}'"
+else
+    sourceDirectory=$PWD
+    echo "No source directory provided -- the root of the repository ('GITHUB_WORKSPACE' environment variable) will be built: '${sourceDirectory}'"
+fi
+
+echo
+oryxCommand="oryx build ${sourceDirectory}"
+echo
+
+if [ -n "${platform}" ]
+then
+    echo "Platform provided: '${platform}'"
+    oryxCommand="${oryxCommand} --platform ${platform}"
+else
+    echo "No platform provided -- Oryx will enumerate the source directory to determine the platform."
+fi
+
+echo
+
+if [ -n "${platformVersion}" ]
+then
+    echo "Platform version provided: '${platformVersion}'"
+    oryxCommand="${oryxCommand} --platform-version ${platformVersion}"
+else
+    echo "No platform version provided -- Oryx will determine the version."
+fi
+
+echo
+echo "Running command '${oryxCommand}'"
+
+echo "GITHUB_RUN_ID is ${GITHUB_RUN_ID}."
+echo "GITHUB_REPOSITORY is ${GITHUB_REPOSITORY}."
+
+url="https://api.github.com/repos/${GITHUB_REPOSITORY}/actions/runs/${GITHUB_RUN_ID}/jobs"
+
+json=$(curl -X GET "${url}")
+
+#Gets the started time and completed time for building container within a Github Action.
+startTime=$(curl -X GET "${url}" | sed 's/,/\n/g' | grep "started_at" | awk '{print $2}' | sed -n '3p')
+endTime=$(curl -X GET "${url}" | sed 's/,/\n/g' | grep "completed_at" | awk '{print $2}' | sed -n '3p')
+
+echo "Started time is ${startTime}."
+echo "Completed time is ${endTime}."
+
+#pass time data as two parameters (or as environment variables?)
+#oryxCommand="${oryxCommand} --github-build-start ${startTime} --github-build-end ${endTime}"
+
+eval $oryxCommand
```

Comment:
I am confused by what does `Completed time` here means. How can we get the completed time even though the build has not happened yet?

Reply:
The completed time (format example: "2020-02-15T02:51:50.000Z") is the time when GitHub action finished downloading images from mcr.microsoft.com/oryx/build:latest. This entrypoint.sh will run afterwards.

Reply:
Ah gotcha. Thanks for clarifying. I think the name and description of the options should clearly say that. Currently it says `build` which is kind of misleading.

Reply:
How about "GITHUB_BUILDIMAGE_START" and "GITHUB_BUILDIMAGE_COMPLETE"? Also, I'm trying to change them to be env variables.

Reply:
IMO environment variable will be better than build parameter, essentially we will be checking the value of the environment variables from oryx to send the calculated value in the telemetry. environment variables or build parameters both works, but imo env variable will help us not to overcrowd oryx build with  parameters that has a limited scope

Reply:
@arroyc Thanks for the advice. Env variable will save us from having too much CLI parameters. I'll go for env variables.